src/pages/ServiceGraph/SummaryPanelCommon.ts

```diff
@@ -29,11 +46,56 @@ export const updateHealth = (summaryTarget: any, stateSetter: (hs: HealthState)
   }
 };
 
-export const nodeData = (node: any) => {
+export const nodeData = (node: any): NodeData => {
   return {
     namespace: node.data('namespace'),
     app: node.data('app'),
     version: node.data('version'),
-    workload: node.data('workload')
+    workload: node.data('workload'),
+    nodeType: node.data('nodeType'),
+    hasParent: !!node.data('parent')
   };
 };
+
+export const getNodeMetricType = (node: any) => {
+  const data = nodeData(node);
+  if (data.nodeType === NodeType.WORKLOAD || (data.nodeType === NodeType.APP && data.hasParent)) {
+    return NodeMetricType.WORKLOAD;
+  } else if (data.nodeType === NodeType.APP) {
+    return NodeMetricType.APP;
+  }
+  return null;
```

Comment:
iirc we want to favor "undefined" over "null", or did the rule change?

Reply:
I don't think the rule changed, I'm going to update to use "undefined"